src/dbnode/integration/fs_bootstrap_volumed_test.go

```diff
@@ -0,0 +1,143 @@
+	if testing.Short() {
+		t.SkipNow() // Just skip if we're doing a short run
+	}
+
+	var (
+		blockSize = 2 * time.Hour
+		rOpts     = retention.NewOptions().SetRetentionPeriod(2 * time.Hour).SetBlockSize(blockSize)
+	)
+	ns1, err := namespace.NewMetadata(testNamespaces[0], namespace.NewOptions().SetRetentionOptions(rOpts))
+	require.NoError(t, err)
+	ns2, err := namespace.NewMetadata(testNamespaces[1], namespace.NewOptions().SetRetentionOptions(rOpts))
+	require.NoError(t, err)
+
+	opts := newTestOptions(t).
+		SetNamespaces([]namespace.Metadata{ns1, ns2})
+	if setTestOpts != nil {
+		opts = setTestOpts(t, opts)
+		ns1 = opts.Namespaces()[0]
+		ns2 = opts.Namespaces()[1]
+	}
+
+	// Test setup
+	setup, err := newTestSetup(t, opts, nil)
+	require.NoError(t, err)
+	defer setup.close()
+
+	fsOpts := setup.storageOpts.CommitLogOptions().FilesystemOptions()
+
+	persistMgr, err := persistfs.NewPersistManager(fsOpts)
+	require.NoError(t, err)
+
+	noOpAll := bootstrapper.NewNoOpAllBootstrapperProvider()
+	bsOpts := result.NewOptions().
+		SetSeriesCachePolicy(setup.storageOpts.SeriesCachePolicy())
+	bfsOpts := fs.NewOptions().
+		SetResultOptions(bsOpts).
+		SetFilesystemOptions(fsOpts).
+		SetDatabaseBlockRetrieverManager(setup.storageOpts.DatabaseBlockRetrieverManager()).
+		SetPersistManager(persistMgr)
+	bs, err := fs.NewFileSystemBootstrapperProvider(bfsOpts, noOpAll)
+	require.NoError(t, err)
+	processOpts := bootstrap.NewProcessOptions().
+		SetTopologyMapProvider(setup).
+		SetOrigin(setup.origin)
+	processProvider, err := bootstrap.NewProcessProvider(bs, processOpts, bsOpts)
+	require.NoError(t, err)
+
+	setup.storageOpts = setup.storageOpts.
+		SetBootstrapProcessProvider(processProvider)
+
+	// Write test data
+	now := setup.getNowFn()
+	inputData1 := []generate.BlockConfig{
+		{IDs: []string{"foo", "bar"}, NumPoints: 100, Start: now.Add(-blockSize)},
+		{IDs: []string{"foo", "baz"}, NumPoints: 50, Start: now},
+	}
+	inputData2 := []generate.BlockConfig{
+		{IDs: []string{"cold1", "cold2"}, NumPoints: 70, Start: now.Add(-blockSize)},
+		{IDs: []string{"cold1", "cold3"}, NumPoints: 20, Start: now},
+	}
+	inputData3 := []generate.BlockConfig{
+		{IDs: []string{"cold4", "cold5", "cold6"}, NumPoints: 30, Start: now.Add(-2 * blockSize)},
+	}
+	allInputData := append(inputData1, inputData2...)
+	allInputData = append(allInputData, inputData3...)
+	if updateInputConfig != nil {
+		updateInputConfig(allInputData)
+	}
+	seriesMaps1 := generate.BlocksByStart(inputData1)
+	seriesMaps2 := generate.BlocksByStart(inputData2)
+	seriesMaps3 := generate.BlocksByStart(inputData3)
+	allSeriesMaps := generate.BlocksByStart(allInputData)
+	require.NoError(t, writeTestDataToDisk(ns1, setup, seriesMaps1, 0))
+	// Write non-zero volumes to disk to simulate a past cold flush.
+	require.NoError(t, writeTestDataToDisk(ns1, setup, seriesMaps2, 1))
```

Comment:
This is interesting, won't you have two files on disk now for same block start? i.e. volume 0 and volume 1 when the node starts up. 

Does it correctly delete volume 0 eventually?

Reply:
Hmm, in debugging this, I realized this test was incorrect, since the multiple `writeTestDataToDisk` doesn't handle any merge logic and so it was losing data... I'll revisit this test another time.

However, this debugging also uncovered a bug when bootstrapping the cold flush version, which I have fixed and added a unit test for.